static/js/click_handlers.js

```diff
@@ -211,6 +211,16 @@ exports.initialize = function () {
         e.stopPropagation();
         e.preventDefault();
     });
+    $("body").on("click", ".reveal_hidden_message", function (e) {
+        const row = $(this).closest(".message_row");
+        const message_id = rows.id_for_recipient_row(row);
+        const message = current_msg_list.get(message_id);
+        current_msg_list.view.rerender_messages([message], false, true);
+        if (current_msg_list === message_list.narrowed) {
+            home_msg_list.view.rerender_messages([message], false, true);
+        }
```

Comment:
Thanks for pointing this out!
I haven't discussed this point specifically with anyone or on CZO, but this probably isn't intended.
The behaviour currently is-
1. In a narrow, reveal the message.
2. Click on the narrow again from the left sidebar, which will rerender it, and the message will be "hidden" again.
3. Click on "All messages", but here, the message is still "revealed"

and, independently,
1. In the home view, reveal the message
2. Narrow to some topic/stream - it is "hidden" again
3. Go back to the home view - it is still "reveal"-ed there.

Since we don't rerender the home view, as you pointed out, the temporary "reveal" mechanism breaks ☹️ 
I think this is what could be done here-
1. On "reveal"-ing from a narrow - reveal the message in the narrow only, and not in the home view. This will work as expected.
2. On "reveal"-ing from the home view, - we can either
  i.  Not do anything on click. This will feel weird to the user, but will prevent the message from staying "reveal"-ed in the home view until the next tab refresh.
  ii. Reveal the message. It will stay revealed in the home view, but users won't feel like the button is broken.

I think we should definitely do (1), but I'm not sure about which option to go with between 2(i) and 2(ii). What are your thoughts on this?

Reply:
https://chat.zulip.org/#narrow/stream/2-general/topic/Mute.20user.20.2316915/near/1123931